opencti-platform/opencti-front/src/private/components/workspaces/WorkspaceDuplicationDialog.tsx

```diff
@@ -0,0 +1,158 @@
+import DialogActions from '@mui/material/DialogActions';
+import DialogTitle from '@mui/material/DialogTitle';
+import TextField from '@mui/material/TextField';
+import { graphql, useMutation } from 'react-relay';
+import { Link } from 'react-router-dom';
+import {
+  WorkspaceDuplicationDialogDuplicatedWorkspaceCreationMutation,
+  WorkspaceDuplicationDialogDuplicatedWorkspaceCreationMutation$data,
+} from '@components/workspaces/__generated__/WorkspaceDuplicationDialogDuplicatedWorkspaceCreationMutation.graphql';
+import { RecordSourceSelectorProxy } from 'relay-runtime';
+import { useFormatter } from '../../../components/i18n';
+import Transition from '../../../components/Transition';
+import { handleError, MESSAGING$ } from '../../../relay/environment';
+
+interface WorkspaceDuplicationDialogProps {
+  workspace: {
+    name: string;
+    type: string;
+    description: string;
+    manifest: string;
+  };
+  displayDuplicate: boolean;
+  duplicating: boolean;
+  handleCloseDuplicate: () => void;
+  setDuplicating: (value: boolean) => void;
+  updater?: (store: RecordSourceSelectorProxy<WorkspaceDuplicationDialogDuplicatedWorkspaceCreationMutation$data>) => void;
+  paginationOptions?: {
+    search: string;
+    orderBy: string;
+    orderMode: string;
+    filters: Array<{ key: string, values: Array<string> }>;
+  };
+}
+
+const workspaceDuplicationDialogDuplicatedWorkspaceCreation = graphql`
+  mutation WorkspaceDuplicationDialogDuplicatedWorkspaceCreationMutation(
+    $input: WorkspaceDuplicateInput!
+  ) {
+    workspaceDuplicate(input: $input) {
+      id
+      ...WorkspaceLine_node
+    }
+  }
+`;
+const WorkspaceDuplicationDialog: FunctionComponent<
+WorkspaceDuplicationDialogProps
+> = ({
+  workspace,
+  duplicating,
+  setDuplicating,
+  displayDuplicate,
+  handleCloseDuplicate,
+  updater,
+  paginationOptions,
+}) => {
+  const { t } = useFormatter();
+  const duplicatedDashboardInitialName = useMemo(() => `${workspace.name} - ${t('copy')}`, [t, workspace.name]);
+  const [newName, setNewName] = useState(duplicatedDashboardInitialName);
+  const [commitDuplicatedWorkspaceCreation] = useMutation<WorkspaceDuplicationDialogDuplicatedWorkspaceCreationMutation>(
+    workspaceDuplicationDialogDuplicatedWorkspaceCreation,
+  );
+
+  const submitDashboardDuplication = (submittedWorkspace: {
+    name: string;
+    type: string;
+    description: string;
+    manifest: string;
+  }) => {
+    commitDuplicatedWorkspaceCreation({
+      variables: {
+        input: {
+          name: submittedWorkspace.name,
+          type: submittedWorkspace.type,
+          description: submittedWorkspace.description,
+          manifest: submittedWorkspace.manifest,
+        },
+      },
+      updater: (store) => updater && updater(store),
+      onError: (error) => {
+        handleError(error);
+      },
+      onCompleted: (data) => {
+        handleCloseDuplicate();
+        if (!paginationOptions) {
```

Comment:
why do we need this check?

Reply:
The mutation was launched from the dashboards list if there are pagination options. In this case, the user is redirected to the duplicated dashboard.

Otherwise, the request was launched from the dashboard view. Then, the user decides whether to stay on the current dashboard view or navigate to the duplicated dashboard.

Would you like it to be explicit, @marieflorescontact ?

Reply:
Yes I think it would be more explicit with a variable